File: src-tauri/src/models/task.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Task {
    pub t_id: i64,
    pub title: String,
    pub due_date: String,
    pub completed: i64,
    pub created_at: String,
    pub updated_at: String,
    pub s_id: i64,
    pub c_id: i64,
}

File: src/services/task.ts
import { invoke } from "@tauri-apps/api/core";
import type { Task } from "../types/task";

export function getTask(): Promise<Task[]>{
    return invoke("get_task");
}

export function createTask(title: string, dueDate: string, completed: number, sId: number, cId: number): Promise<void>{
    return invoke("create_task", {title, dueDate, completed, sId, cId })
}

export function updateTask(tId: number, title: string, dueDate: string, completed: number, sId: number, cId: number): Promise<Task>{
    return invoke("update_task", {tId, title, dueDate, completed, sId, cId })
}

export function deleteTask(tId: number): Promise<void>{
    return invoke("delete_task", {tId})
}